app/router/session_occurrence.py

```diff
@@ -0,0 +1,114 @@
+from fastapi import APIRouter, HTTPException, Request
+import requests
+from settings import settings
+from datetime import datetime
+from models import SessionResponse
+
+router = APIRouter(prefix="/session-occurrence", tags=["Session Occurrence"])
+session_db_url = settings.db_url + "/session/"
+session_occurrence_db_url = settings.db_url + "/session-occurrence/"
+
+
+def build_date_and_time(date_time: str):
+    """
+    Parses string into date and time strings
+    """
+    parsed_time = datetime.strptime(date_time, "%Y-%m-%dT%H:%M:%SZ")
+    date_string, time_string = str(parsed_time).split()
+    return (date_string, time_string)
+
+
+def get_current_datetime():
+    """
+    Returns current datetime
+    """
+    current_datetime = datetime.now()
+    return build_date_and_time(current_datetime.strftime("%Y-%m-%dT%H:%M:%SZ"))
+
+
+def has_session_started(start_time: str):
+    """
+    Checks if session has started
+    - If session time is given and session start date is less than or equal to current date, returns True
+    - Otherwise, returns False
+    """
+    if start_time is not None:
+        session_start_date, session_start_time = build_date_and_time(start_time)
+        current_date, current_time = get_current_datetime()
+        return session_start_date == current_date and session_start_time <= current_time
+    return True
+
+
+def has_session_ended(end_time: str):
+    """
+    Checks if session has ended
+    - If end time is given and if session end date is greater than or equal to current date, returns True
+    - Else, always returns True
+    """
+    if end_time is not None:
+        session_end_date, session_end_time = build_date_and_time(end_time)
+        current_date, current_time = get_current_datetime()
+        return session_end_date == current_date and session_end_time >= current_time
+    return True
+
+
+@router.get("/", response_model=SessionResponse)
+async def get_session_occurrence_data(request: Request):
+    """
+    This API returns session occurrence details corresponding to the provided session ID, if the ID exists in the database and if the session is open
+
+    Parameters:
+    session_id (str): The ID against which details need to be retrieved
+
+    Returns:
+    list: Returns session occurrence details if the session ID exists in the database and the session is open, false if the session is closed. If the ID does not exist, 404 is returned
+
+    Example:
+    > $BASE_URL/session/1234
+    returns [{session_data}]
+
+    > $BASE_URL/session/{closed_session_id}
+    returns {"is_session_open": False}
```

Comment:
Why not return the session date in addition to returning that it is closed?

Reply:
Returning session data requires an additional API call. We don't really use session data if the session is closed.